File: Prog2/HelloLiskov/Liskov.java
class Teglalap
{
	protected int m_szelesseg;
	protected int m_magassag;
	public void setSzelesseg(int szelesseg){
	m_szelesseg = szelesseg;
	}
	public void setMagassag(int magassag){
		m_magassag = magassag;
		}
	public int getSzelesseg(){
		return m_szelesseg;
		}
	public int getMagassag(){
		return m_magassag;
		}
	public int getTerulet(){
		return m_szelesseg * m_magassag;
		}
}
class Negyzet extends Teglalap
{
	public void setSzelesseg(int szelesseg){
		m_szelesseg = szelesseg;
		m_magassag = szelesseg;
		}
	public void setMagassag(int magassag){
		m_szelesseg = magassag;
		m_magassag = magassag;
		}
}
class LspTest
{
	private static Teglalap getUjTeglalap()
	{
	return new Negyzet();
	}
	public static void main (String args[])
	{
		Teglalap r = LspTest.getUjTeglalap();
		r.setSzelesseg(5);
		r.setMagassag(10);
		System.out.println(r.getTerulet());
	}
}

File: Prog2/HelloLiskov/Liskov.cpp
#include <iostream>

class Teglalap
{
	public:
		int m_szelesseg;
		int m_magassag;
		virtual void setszelesseg(int szelesseg){
			m_szelesseg = szelesseg;
			}
		virtual void setmagassag(int magassag){
		m_magassag = magassag;
		}
		int getszelesseg(){
			return m_szelesseg;
		}
		int getmagassag(){
			return m_magassag;
		}
		int getTerulet(){
			return m_szelesseg * m_magassag;
		}
};

class Negyzet : public Teglalap
{
	public:
		void setszelesseg(int szelesseg){
			m_szelesseg = szelesseg;
			m_magassag = szelesseg;
			}
		void setmagassag(int magassag){
			m_szelesseg = magassag;
			m_magassag = magassag;
			}
};

Teglalap* getNewTeglalap()
{
	return new Negyzet();
}

int main ()
{
	Teglalap* r = getNewTeglalap();
	r->setszelesseg(5);
	r->setmagassag(10);
	std::cout << r->getTerulet();
}